### PR TITLE
I returned change to the sender wallet

### PR DESCRIPTION
**Pull request by ak-toktor**

### CODE REVIEW

**Comment by vadymeng0** [comment]
Could you please provide below things in the description?
    - What is this task about, and related Freedcamp task.
    - Test result.
    - (If possible) some short description on how you implement it.
 Thank you!!!

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -3981,14 +3981,14 @@ UniValue z_getoperationstatus_IMPL(const UniValue& params, bool fRemoveFinishedO
 #define CTXIN_SPEND_DUST_SIZE   148
 #define CTXOUT_REGULAR_SIZE     34
 
-UniValue z_sendmany(const UniValue& params, bool fHelp)
+UniValue z_sendmanyimpl(const UniValue& params, bool fHelp, const std::string& functionName, bool returnChangeToSenderAddr)
```
Comment (vadymeng0) [inline src/wallet/rpcwallet.cpp:3984]: These two parameter has the same meaning. Should we deduct one?

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -4249,6 +4245,23 @@ UniValue z_sendmany(const UniValue& params, bool fHelp)
         contextualTx.nVersion = 2; // Tx format should support vjoinsplits 
     }
 
+    // push sender address to the recipients to return change back to  the sender 
+    if(returnChangeToSenderAddr) {
+        int nMinDepth = 1;
+        if (fromTaddr) {
+            CAmount nBalance = getBalanceTaddr(fromaddress, nMinDepth, false);
```
Comment (vadymeng0) [inline src/wallet/rpcwallet.cpp:4252]: In the case we return change to sender address, a new operation will still be added to the global queue? Is that mean another transaction will be created?

File: src/wallet/rpcwallet.cpp [hunk 3]
```diff
@@ -5058,6 +5080,7 @@ static const CRPCCommand commands[] =
     { "wallet",             "z_gettotalbalance",        &z_gettotalbalance,        false },
     { "wallet",             "z_mergetoaddress",         &z_mergetoaddress,         false },
     { "wallet",             "z_sendmany",               &z_sendmany,               false },
+    { "wallet",             "z_sendmanywithchangetosender", &z_sendmanywithchangetosender, false },
```
Comment (vadymeng0) [inline src/wallet/rpcwallet.cpp:5083]: This is **NOT** following the latest requirement that is suggested by @a-ok123 and @Dicklesworthstone 
We need to add _additional flag_ to `z_sendmany`, so the implementation must support:
    - `z_sendmany` (without additional flag): just behave old implementation. 
    - `z_sendmany -changetosendingaddress`: do the change utxo change back to the old address.
But you are creating a brand new command `z_sendmanywithchangetosender`.

**Comment by vadymeng0** [comment]
This is reimplemented in: https://github.com/pastelnetwork/pastel/pull/51

**Comment by vadymeng0** [comment]
I close this PR be cause it basically do not solve the problem.

**Review by vadymeng0 — CHANGES_REQUESTED** [review]
Basically this change is adding more transaction to send the change back to the sender, but it **_do NOT remove the part that new address is generated and the change is sending to that new address_**.

**Comment by a-ok123** [comment]
replaced by PR51

**Pull request closed without merging**